agent/nrpe.rb

```diff
@@ -62,16 +63,25 @@ def self.run(command)
         return exitcode, output
       end
 
-      def self.plugin_for_command(command)
+      def self.plugin_for_command(command, *args)
         plugins = Nrpe.all_command_plugins
 
         if plugins.include?(command)
-          return plugins[command]
+          return self.parse_args_on_command(plugins[command], *args)
         end
 
         return nil
       end
 
+      def self.parse_args_on_command(command, *args)
+        arguments = []
+        args.to_enum.with_index.each do |arg, i|
+          arguments.push(["$ARG#{i+1}$","#{arg}"])
+        end
+        arguments.each {|arg| command.gsub!(arg[0], arg[1])}
+        return command
+      end
```

Comment:
This is a really quite weird way to write:

```
def self.expand_command_args(command, args)
  copy = command.dup # work on a copy to avoid mutating our arguments
  args.each_with_index do |value, index|
    copy.gsub!(/\$ARG#{index + 1}\$/, value)
  end
  return copy
end
```


Also `parse_args_on_command` is a poor name for a method.  Maybe `expand_command_args` would be more apt.